<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="ie=edge"/><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/><style data-href="/styles.f6f5281c6d60e07d9c77.css" id="gatsby-global-css">@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:200;src:local("Nunito Sans Extra Light "),local("Nunito Sans-Extra Light"),url(/static/nunito-sans-latin-200-3f69f8c78eb965be709fada6e9326097.woff2) format("woff2"),url(/static/nunito-sans-latin-200-0aa9029aa5c94ee0245428787ff2aa8b.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:200;src:local("Nunito Sans Extra Light italic"),local("Nunito Sans-Extra Lightitalic"),url(/static/nunito-sans-latin-200italic-6efd102c0b004908b9454b4e4610d693.woff2) format("woff2"),url(/static/nunito-sans-latin-200italic-da7c09610c91ac990644a19f2d713357.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:300;src:local("Nunito Sans Light "),local("Nunito Sans-Light"),url(/static/nunito-sans-latin-300-f16977f35222a4cce7d6f4336732a67a.woff2) format("woff2"),url(/static/nunito-sans-latin-300-e7a891b0ad0ce9050b19e3d0cdfcd860.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:300;src:local("Nunito Sans Light italic"),local("Nunito Sans-Lightitalic"),url(/static/nunito-sans-latin-300italic-246f6e2b70f0b23e366947c975effa65.woff2) format("woff2"),url(/static/nunito-sans-latin-300italic-1a3d2422361712bf569edf8b2e1f6d90.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:400;src:local("Nunito Sans Regular "),local("Nunito Sans-Regular"),url(/static/nunito-sans-latin-400-9c6a1ff29aaba0c21496eac04be062df.woff2) format("woff2"),url(/static/nunito-sans-latin-400-8ace450709844437a579d4f4c7c37b42.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:400;src:local("Nunito Sans Regular italic"),local("Nunito Sans-Regularitalic"),url(/static/nunito-sans-latin-400italic-d44697ce7f31cee44f701572fb38cab5.woff2) format("woff2"),url(/static/nunito-sans-latin-400italic-e14ffefb3a60e5d5a9a47519882f92e8.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:600;src:local("Nunito Sans SemiBold "),local("Nunito Sans-SemiBold"),url(/static/nunito-sans-latin-600-b56c7e1dd3d7a5fad61e46f4dd553294.woff2) format("woff2"),url(/static/nunito-sans-latin-600-5a71131c00b436beb97dd2c570768f05.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:600;src:local("Nunito Sans SemiBold italic"),local("Nunito Sans-SemiBolditalic"),url(/static/nunito-sans-latin-600italic-0de4a8bace81fc8b1f570af06a28e385.woff2) format("woff2"),url(/static/nunito-sans-latin-600italic-9db53e390a5327462f45889e249f84d6.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:700;src:local("Nunito Sans Bold "),local("Nunito Sans-Bold"),url(/static/nunito-sans-latin-700-d0b6d0f05c00552a21519d6bdb3e5ed7.woff2) format("woff2"),url(/static/nunito-sans-latin-700-1aa5d63eca4940fbbb5181895e9bc008.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:700;src:local("Nunito Sans Bold italic"),local("Nunito Sans-Bolditalic"),url(/static/nunito-sans-latin-700italic-c7569232f38ac2b1b3211151127b1ed3.woff2) format("woff2"),url(/static/nunito-sans-latin-700italic-ffc1f5c22dc4b08f710c027976972de1.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:800;src:local("Nunito Sans ExtraBold "),local("Nunito Sans-ExtraBold"),url(/static/nunito-sans-latin-800-6a4c29ad18bd8072a9a9d2443d0e340e.woff2) format("woff2"),url(/static/nunito-sans-latin-800-a28845a6196298dad6dc0b2ba0b6de55.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:800;src:local("Nunito Sans ExtraBold italic"),local("Nunito Sans-ExtraBolditalic"),url(/static/nunito-sans-latin-800italic-0117d6315806b392f7da0d5d3aa9c7ca.woff2) format("woff2"),url(/static/nunito-sans-latin-800italic-4f4e7a921cc52e9458245f66eb550004.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:normal;font-display:swap;font-weight:900;src:local("Nunito Sans Black "),local("Nunito Sans-Black"),url(/static/nunito-sans-latin-900-e22a661eeed6baa960d645c48de5a00f.woff2) format("woff2"),url(/static/nunito-sans-latin-900-412c6f01c1eb3a9135d53bc4fd220a35.woff) format("woff")}@font-face{font-family:Nunito Sans;font-style:italic;font-display:swap;font-weight:900;src:local("Nunito Sans Black italic"),local("Nunito Sans-Blackitalic"),url(/static/nunito-sans-latin-900italic-0dc1055cf601fe66514482613ebdf168.woff2) format("woff2"),url(/static/nunito-sans-latin-900italic-708845e875a3b08640369bf84fd65472.woff) format("woff")}ul{line-height:1}ul.uncompressed{line-height:normal}ul.compressed{line-height:1!important}span[title]{-webkit-text-decoration-line:underline;text-decoration-line:underline;-webkit-text-decoration-style:dotted;text-decoration-style:dotted}span[title].no-title-dash{-webkit-text-decoration-line:none;text-decoration-line:none;-webkit-text-decoration-style:none;text-decoration-style:none}.section-link{font-weight:lighter;float:right}.section-link:hover{font-weight:700}html{font-family:sans-serif;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}article,aside,details,figcaption,figure,footer,header,main,menu,nav,section,summary{display:block}audio,canvas,progress,video{display:inline-block}audio:not([controls]){display:none;height:0}progress{vertical-align:baseline}[hidden],template{display:none}a{background-color:transparent;-webkit-text-decoration-skip:objects;color:#5f160c;text-decoration:none}a:active,a:hover{outline-width:0}a:hover,abbr[title]{text-decoration:underline}abbr[title]{border-bottom:none;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}b,strong{font-weight:inherit;font-weight:bolder}dfn{font-style:italic}h1{font-size:2em;margin:.67em 0}mark{background-color:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}img{border-style:none}svg:not(:root){overflow:hidden}code,kbd,pre,samp{font-family:monospace,monospace;font-size:1em}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}button,input,optgroup,select,textarea{font:inherit;margin:0}optgroup{font-weight:700}button,input{overflow:visible}button,select{text-transform:none}[type=reset],[type=submit],button,html [type=button]{-webkit-appearance:button}[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{border-style:none;padding:0}[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{outline:1px dotted ButtonText}fieldset{border:1px solid silver;margin:0 2px;padding:.35em .625em .75em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}textarea{overflow:auto}[type=checkbox],[type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{-webkit-appearance:none}::-webkit-input-placeholder{color:inherit;opacity:.54}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}html{font:112.5%/1.45em georgia,serif;box-sizing:border-box;overflow-y:scroll}*,:after,:before{box-sizing:inherit}body{color:rgba(0,0,0,.8);font-family:Nunito Sans,Noto Sans,Lucida Sans Unicode,Lucida Sans,Tahoma,sans-serif;font-weight:400;word-wrap:break-word;-webkit-font-kerning:normal;font-kerning:normal;-ms-font-feature-settings:"kern","liga","clig","calt";font-feature-settings:"kern","liga","clig","calt"}img{max-width:100%;padding:0;margin:0 0 1.45rem}h1{font-size:2.25rem}h1,h2{padding:0;margin:0 0 1.45rem;color:inherit;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;font-weight:700;text-rendering:optimizeLegibility;line-height:1.1}h2{font-size:1.62671rem}h3{font-size:1.38316rem}h3,h4{padding:0;margin:0 0 1.45rem;color:inherit;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;font-weight:700;text-rendering:optimizeLegibility;line-height:1.1}h4{font-size:1rem}h5{font-size:.85028rem}h5,h6{padding:0;margin:0 0 1.45rem;color:inherit;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;font-weight:700;text-rendering:optimizeLegibility;line-height:1.1}h6{font-size:.78405rem}hgroup{padding:0;margin:0 0 1.45rem}ol,ul{padding:0;margin:0 0 1.45rem 1.45rem;list-style-position:outside;list-style-image:none}dd,dl,figure,p{padding:0;margin:0 0 1.45rem}pre{margin:0 0 1.45rem;font-size:.85rem;line-height:1.42;background:rgba(0,0,0,.04);border-radius:3px;overflow:auto;word-wrap:normal;padding:1.45rem}table{font-size:1rem;line-height:1.45rem;border-collapse:collapse;width:100%}fieldset,table{padding:0;margin:0 0 1.45rem}blockquote{padding:0;margin:0 1.45rem 1.45rem}form,iframe,noscript{padding:0;margin:0 0 1.45rem}hr{padding:0;margin:0 0 calc(1.45rem - 1px);background:rgba(0,0,0,.2);border:none;height:1px}address{padding:0;margin:0 0 1.45rem}b,dt,strong,th{font-weight:700}li{margin-bottom:.725rem}ol li,ul li{padding-left:0}li>ol,li>ul{margin-left:1.45rem;margin-bottom:.725rem;margin-top:.725rem}blockquote :last-child,li :last-child,p :last-child{margin-bottom:0}li>p{margin-bottom:.725rem}code,kbd,samp{font-size:.85rem;line-height:1.45rem}abbr,abbr[title],acronym{border-bottom:1px dotted rgba(0,0,0,.5);cursor:help}abbr[title]{text-decoration:none}td,th,thead{text-align:left}td,th{border-bottom:1px solid rgba(0,0,0,.12);font-feature-settings:"tnum";-moz-font-feature-settings:"tnum";-ms-font-feature-settings:"tnum";-webkit-font-feature-settings:"tnum";padding:.725rem .96667rem calc(.725rem - 1px)}td:first-child,th:first-child{padding-left:0}td:last-child,th:last-child{padding-right:0}code,tt{background-color:rgba(0,0,0,.04);border-radius:3px;font-family:SFMono-Regular,Consolas,Roboto Mono,Droid Sans Mono,Liberation Mono,Menlo,Courier,monospace;padding:.2em 0}pre code{background:none;line-height:1.42}code:after,code:before,tt:after,tt:before{letter-spacing:-.2em;content:" "}pre code:after,pre code:before,pre tt:after,pre tt:before{content:""}@media only screen and (max-width:480px){html{font-size:100%}}.index-module--sameAsList--hXj0k,.index-module--timelineList--2EpGf{line-height:1}.projects-module--projectTitle--154kd{font-size:120%}hr.projects-module--projectEndHR--28ObJ{overflow:visible;padding:0;border:none;border-top:double #333;color:#333;text-align:center}hr.projects-module--projectEndHR--28ObJ:after{display:inline-block;position:relative;top:-.7em;font-size:1.5em;padding:0 .25em;background:#fff}.timeline-module--timelineEvent--1GFNQ{margin-bottom:1em}.timeline-module--timelineEvent--1GFNQ p{padding-left:1.5em}</style><meta name="generator" content="Gatsby 2.31.1"/><title data-react-helmet="true">Timeline | Gaurav Vaidya</title><meta data-react-helmet="true" name="description" content="This is Gaurav Vaidya&#x27;s homepage, with everything you ever wanted to know about him but were afraid to ask."/><meta data-react-helmet="true" property="og:title" content="Timeline"/><meta data-react-helmet="true" property="og:description" content="This is Gaurav Vaidya&#x27;s homepage, with everything you ever wanted to know about him but were afraid to ask."/><meta data-react-helmet="true" property="og:type" content="website"/><meta data-react-helmet="true" name="twitter:card" content="summary"/><meta data-react-helmet="true" name="twitter:creator" content="@gaurav"/><meta data-react-helmet="true" name="twitter:title" content="Timeline"/><meta data-react-helmet="true" name="twitter:description" content="This is Gaurav Vaidya&#x27;s homepage, with everything you ever wanted to know about him but were afraid to ask."/><link rel="icon" href="/favicon-32x32.png?v=feefe1922fc7f16cabcc0a7d26c91eac" type="image/png"/><link rel="manifest" href="/manifest.webmanifest" crossorigin="anonymous"/><meta name="theme-color" content="#663399"/><link rel="apple-touch-icon" sizes="48x48" href="/icons/icon-48x48.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="72x72" href="/icons/icon-72x72.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="96x96" href="/icons/icon-96x96.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="144x144" href="/icons/icon-144x144.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="192x192" href="/icons/icon-192x192.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="256x256" href="/icons/icon-256x256.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="384x384" href="/icons/icon-384x384.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><link rel="apple-touch-icon" sizes="512x512" href="/icons/icon-512x512.png?v=feefe1922fc7f16cabcc0a7d26c91eac"/><script src="https://code.jquery.com/jquery-3.5.1.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><link as="script" rel="preload" href="/webpack-runtime-f41cac52ff50071fe1f4.js"/><link as="script" rel="preload" href="/framework-acd7498685eeb36e39da.js"/><link as="script" rel="preload" href="/app-1ff00ff6183944037c30.js"/><link as="script" rel="preload" href="/styles-407fe62976dc5310c43e.js"/><link as="script" rel="preload" href="/0f1ac474-cb5adb2ed9d11fe078e1.js"/><link as="script" rel="preload" href="/commons-47785431c9696da9c5f8.js"/><link as="script" rel="preload" href="/0b10d7199d01ceeef934b1f7be1d4d848090e8b1-ce937dcc159ebdddc4a8.js"/><link as="script" rel="preload" href="/f0d9b79f24a0985e3f3911e871f5374fd763bbdf-efc7f1ba0523bbcd09ad.js"/><link as="script" rel="preload" href="/component---src-pages-timeline-js-dcc741760f588e8f0200.js"/><link as="fetch" rel="preload" href="/page-data/timeline/page-data.json" crossorigin="anonymous"/><link as="fetch" rel="preload" href="/page-data/sq/d/1490079655.json" crossorigin="anonymous"/><link as="fetch" rel="preload" href="/page-data/sq/d/3649515864.json" crossorigin="anonymous"/><link as="fetch" rel="preload" href="/page-data/sq/d/63159454.json" crossorigin="anonymous"/><link as="fetch" rel="preload" href="/page-data/app-data.json" crossorigin="anonymous"/></head><body><div id="___gatsby"><div style="outline:none" tabindex="-1" id="gatsby-focus-wrapper"><header style="background:#5f160c;margin-bottom:1.45rem;border-bottom:1px solid #421610"><div style="margin:0 auto;max-width:960px;padding:1.45rem 1.0875rem"><a href="/avatar.jpeg"><div class=" gatsby-image-wrapper" style="position:relative;overflow:hidden;display:inline-block;width:64px;height:64px;float:left;margin-top:-0.5em;margin-right:0.5em;text-align:center"><img aria-hidden="true" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAAFtUlEQVQ4yx2SSWwb9xXG/1SB5NBD0KJFEzhAgKJGXDQGLAWKVSuVLVurLVGytVCSSYn7zuEMyeFwGS7DbUiREndRJCVqcZtYTWKnKIIeeihQ+FQ06aWHwvClQc4p0KNI/r/SPTw84D3g4Xu/7yNWv0zsAZkw8ilx2BxDjNVEgL8rUk7lYsb98FR2P3ols6vdvG+9l+HWvw2aH1wURLPq1ct/vFmun5I/fHkyVD44JaV6hzQ754ToHWFiZmPkCiEKi4UhVoNhKszpv5GcSkjWeWQ9a5CMU5DZFWS5VbyeH+Z5NGv5l+lsde3kvEOk8/+SSrmhaNQbhNj8GeLRbyhm5hYJL+0FLAYDPLZtJLj1Xtw2380Hd/qS8R4VtZNU5jfpLr/WrcS2u6WciNx+HXulRtliZgj+809SLpYUROcSh7adElnWuAMGVwRGo5kylu1u3P2Ipr0DRfZ5Km7fpPzaCA1ujVJBNYwUM0+rWa7XPsz1jk+P0Oyc1TJnX5Fnf/qbgqwsqYlqRTt1994i7i9vUk4Q+5Ewi7BjiYb0t6lbeQ38xii45euD/hH8Gx/SiO7XtJaxobnP0/ah3G2365Bj6Z12oUzIi/xzxc6O9RuHJ4RwWu6mZQnpOEOjnsc0ZJoGs/wBku6BUuscQro7/2eYsE2jVRToUT2BZinYO2rIiPr579auT7xF1Kszi+rHangj8V48k6bJRIBKAQNNBvXIx8wQzfNI8xpkBTVE6yLibhX2YzYct2o4e9Khh8UwWrXY5b4YQMYp2AnrNp1q9Vq4ff5uMhVFIsrSdNSJbMKDciGOQpxFLsYizm7Ap56A9/EEKtkA2odVnD19To9bRbQbcq8oepFkgi+ITrv5ig94EAx7+7EIS6WwE6mwGRnRiXrnU+ynBdRKMlIhG+zLN+DV3Ea1lEa7c4LTL/+CxvkFrReTNO3WDswKXpLbt0a6fo8BvFtNQ7yJRsMMdgsZVOtltJ98hmKjg0Ihi5TEg9XchWBRojbY5Q4+hcrhh5n10Hqrg3yApbLZCTJ8/WpvZ3MRlu1FGvKZqCjYkcoXkaq2wCXKcAbTiO0dIjj4QmA0iPq0aB6dwC5EMTo9AZOTpfVmm8peN5JmhpL3rvz4W+X8beg25+n20jhldUsIxtMYuzOGd9+/huHxMSjX9ZD2jxAPcwMcNhwcVBGIZWG3OgbBrtNiJoSUZYDJ4u6Tn/3ojYuR61exPHure+uX78Bu3KLF1jk2jBZMLMziw7ExjI9NIt/+ApmBaUneiELCj2qlgdbZc1rZzyAT8vVlkwNZo+1r8vMrP1T94r2f4uPx8e7kx6NUOTVKOacByXQSQVGCNyTD509AkjKQBAYpv22Qy03sigyOT45pbT+NSCB+GfDEkWYEP7n50fCbH1x95+WNa+9iZvI3vbnZKSxM3aTmrTlEfFbslesotX6HKM9AdKiRT4jIDo6FtDO0eVhBo/O0n+ATSAip772Vi7fJyPs/eV1rM+PDUCtXeo8fqunqytYAwSw1G3eoXCiidNCGyJrAG9axl9ule4UiTQhOHH/yjB789qvLdHgXeU/Ek/LGCJkevjM0N3KLqO8vlTfur0LzaKdr09h7JhMPKyvBnz2kqeoZzbc+o15OoNLAjGy2jkKl03/y11eXuVwLCSb4eVTHEO7P/yZENT1LKi4vEaIhoppZqqkWVNA+3IZN5770ROs9LndOw9UvsP/Hf9Fo7Rn0G5o+Z7Bc7h48HeT1FFEr97m7cPED0cgSiQkpyPLEBHk4Mam48caviO7+KlFPL+xoVcbvHAYPHLYQrPwubEKRuqJNCMXfw8klYNO6EPImvw9s6D1M5hMS2dKQuIUbSrgEQtbvLZC1uw/Io7sPFE6dW6FdWCf+cOkt3brObts0v7Bvmi4tGjtcazrqVJn6Hj33tU/n9pumlW8HVFqyC5CowaWIGl2vFZL/AbKfUsXOQIwbAAAAAElFTkSuQmCC" alt="" style="position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover;object-position:center;opacity:1;transition-delay:500ms"/><picture><source srcSet="/static/feefe1922fc7f16cabcc0a7d26c91eac/ea22a/avatar-round.png 1x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/88b72/avatar-round.png 1.5x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/2c344/avatar-round.png 2x"/><img srcSet="/static/feefe1922fc7f16cabcc0a7d26c91eac/ea22a/avatar-round.png 1x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/88b72/avatar-round.png 1.5x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/2c344/avatar-round.png 2x" src="/static/feefe1922fc7f16cabcc0a7d26c91eac/ea22a/avatar-round.png" alt="A photo of a cat I use as my avatar image" width="64" height="64" loading="eager" style="position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover;object-position:center;opacity:0;transition:opacity 500ms"/></picture><noscript><picture><source srcset="/static/feefe1922fc7f16cabcc0a7d26c91eac/ea22a/avatar-round.png 1x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/88b72/avatar-round.png 1.5x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/2c344/avatar-round.png 2x" /><img loading="eager" width="64" height="64" srcset="/static/feefe1922fc7f16cabcc0a7d26c91eac/ea22a/avatar-round.png 1x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/88b72/avatar-round.png 1.5x,
/static/feefe1922fc7f16cabcc0a7d26c91eac/2c344/avatar-round.png 2x" src="/static/feefe1922fc7f16cabcc0a7d26c91eac/ea22a/avatar-round.png" alt="A photo of a cat I use as my avatar image" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/></picture></noscript></div></a><h1 style="margin:0"><a style="color:white;text-decoration:none" href="/">Gaurav Vaidya</a></h1></div></header><div style="margin:0 auto;max-width:960px;padding:0 1.0875rem 1.45rem"><h1>Timeline</h1><main><small>(<a href="#">Toggle showing products</a>)</small><p></p><div id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA">§</a><span title="October 21, 2019, 1 year, 5 months and 11 days ago">Oct 2019</span> <span class="no-title-dash" title="1 year, 1 month and 29 days">to</span> <span title="December 20, 2020, 3 months and 12 days ago">Dec 2020</span> <!-- -->•<!-- --> <strong>Semantics web technologist</strong> <!-- -->at<!-- --> <span>the <a href="https://www.renci.org/" title="Located at Chapel Hill, North Carolina, USA">Renaissance Computing Institute (RENCI)</a></span></div><div><div><em>Part of <span>the <a href="https://www.unc.edu/" title="Located at Chapel Hill, North Carolina, USA">University of North Carolina at Chapel Hill</a></span></em></div></div><p>I work on several projects to facilitate the re-use of biomedical data by other scientists, with a particular focus on using Semantic Web technologies to make scientific data findable, accessible, interoperable, and reusable. To achieve this, I work with tools that use standards such as RDF and JSON-LD to store and transmit semantically rich data, use ontologies written in OWL to reason over data, and use descriptions in SHACL and ShEx to validate data and document data models. Where existing tools are unavailable or inadequate, I write software tools to fill this need. My current projects focus on genetic, disease and other biomedical data, but I look forward to working with other kinds of data in the future. I work primarily in Scala to build these tools.</p><p><ul class="uncompressed roleProducts"><li id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Center-for-Cancer-Data-Harmonization-(CCDH)"><a class="section-link" href="#Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Center-for-Cancer-Data-Harmonization-(CCDH)">§</a><strong>Project:</strong> <a href="/projects#Center-for-Cancer-Data-Harmonization-(CCDH)">Center for Cancer Data Harmonization (CCDH)</a> (<span title="October 21, 2019, 1 year, 5 months and 11 days ago">Oct 2019</span> <span class="no-title-dash" title="1 year, 1 month and 29 days">to</span> <span title="December 20, 2020, 3 months and 12 days ago">Dec 2020</span>): <!-- -->I work on the CCDH Tools and Data Quality team at RENCI. We aim to develop software tools that will allow the validation of biomedical data against the CRDC-H data model to store cancer-related biomedical data, to publish and transform the data model into different formats as needed, and to fulfil any other software development tasks that are needed to complete this project.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Center-for-Cancer-Data-Harmonization-(CCDH)-csv2caDSR" href="https://github.com/cancerDHC/csv2caDSR">csv2caDSR</a> (<span title="July 28, 2020, 8 months and 4 days ago">Jul 2020</span> <span class="no-title-dash" title="4 months and 6 days">to</span> <span title="December 4, 2020, 3 months and 28 days ago">Dec 2020</span>)<!-- -->: <!-- -->A tool for harmonizing biomedical data using the Cancer Data Standards Registry and Repository (caDSR) as a source of validation information.</li><li><strong>Software</strong>: <a id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Center-for-Cancer-Data-Harmonization-(CCDH)-UMLS-RRF-Scala" href="https://github.com/cancerDHC/umls-rrf-scala">UMLS-RRF-Scala</a> (<span title="March 30, 2020, 1 year, zero months and 2 days ago">Mar 2020</span> <span class="no-title-dash" title="7 months and 5 days">to</span> <span title="November 4, 2020, 4 months and 28 days ago">Nov 2020</span>)<!-- -->: <!-- -->A set of tools for identifying mappings from terms in one vocabulary to others, intending to be used to provide mappings from closed-source vocabularies to open-access vocabularies.</li></ul><li id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Small-projects-at-RENCI"><a class="section-link" href="#Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Small-projects-at-RENCI">§</a><strong>Project:</strong> <a href="/projects#Small-projects-at-RENCI">Small projects at RENCI</a> (<span title="October 21, 2019, 1 year, 5 months and 11 days ago">Oct 2019</span> <span class="no-title-dash" title="1 year, 1 month and 29 days">to</span> <span title="December 20, 2020, 3 months and 12 days ago">Dec 2020</span>): <!-- -->While working as a Semantic Web technologist at RENCI, I developed a number of ad-hoc software tools to meet various needs from other RENCI teams.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Small-projects-at-RENCI-SHACLI" href="http://shacli.org/">SHACLI</a> (<span title="October 30, 2019, 1 year, 5 months and 2 days ago">Oct 2019</span> <span class="no-title-dash" title="6 months and 26 days">to</span> <span title="May 26, 2020, 10 months and 6 days ago">May 2020</span>)<!-- -->: <!-- -->A command-line interface (CLI) for the Shapes Constraint Language (SHACL) to validate RDF data against SHACL shapes with improved error messages. This was part of a project to simplify data model creation, so that one document can be used to produce both the documentation and can be used to validate input data.</li><li><strong>Software</strong>: <a id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Small-projects-at-RENCI-Omnicorp-and-OmniCORD" href="https://github.com/NCATS-Gamma/omnicorp">Omnicorp and OmniCORD</a> (<span title="October 23, 2019, 1 year, 5 months and 9 days ago">Oct 2019</span> <span class="no-title-dash" title="1 year, zero months and 16 days">to</span> <span title="November 8, 2020, 4 months and 24 days ago">Nov 2020</span>)<!-- -->: <!-- -->I built upon the existing Omnicorp tool to improve the RDF data being produced by it, such as authorship information. I also wrote OmniCORD, a variant of Omnicorp for extracting entities from the COVID-19 Open Research Dataset (CORD-19).</li><li><a class="section-link" href="#Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Small-projects-at-RENCI-https:doi.org10.1093bioinformaticsbtaa718">§</a><strong id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-Small-projects-at-RENCI-https:doi.org10.1093bioinformaticsbtaa718">Citation</strong>: <a href="http://orcid.org/0000-0002-1780-9872">Daniel Korn</a>, <!-- -->Tesia Bobrowski<!-- -->, <!-- -->Michael Li<!-- -->, <!-- -->Yaphet Kebede<!-- -->, <!-- -->Patrick Wang<!-- -->, <!-- -->Phillips Owen<!-- -->, <strong><em>Gaurav Vaidya</em></strong>, <!-- -->Eugene Muratov<!-- -->, <!-- -->Rada Chirkova<!-- -->, <!-- -->Chris Bizon<!-- -->, <a href="http://orcid.org/0000-0003-3802-8896">Alexander Tropsha</a> <!-- -->(<!-- -->November 11, 2020<!-- -->)<!-- --> <a href="https://doi.org/10.1093/bioinformatics/btaa718">COVID-KOP: integrating emerging COVID-19 data with the ROBOKOP database</a>.<!-- --> <em>Bioinformatics</em> <strong></strong>.</li></ul><li><a class="section-link" href="#Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-My-page-on-the-RENCI-website">§</a><strong>Link</strong>: <a id="Semantics-web-technologist-the-Renaissance-Computing-Institute-(RENCI)-at-Chapel-Hill-North-Carolina-USA-My-page-on-the-RENCI-website" href="https://renci.org/staff/gaurav-vaidya/">My page on the RENCI website</a></li></ul></p></div><div id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA">§</a><span title="January 2, 2018, 3 years, 3 months and 1 day ago">Jan 2018</span> <span class="no-title-dash" title="1 year, 9 months and 9 days">to</span> <span title="October 11, 2019, 1 year, 5 months and 21 days ago">Oct 2019</span> <!-- -->•<!-- --> <strong>Postdoctoral associate</strong> <!-- -->at<!-- --> <span>the <a href="https://www.floridamuseum.ufl.edu/" title="Located at Gainesville, Florida, USA">Florida Museum of Natural History</a></span></div><div><div><em>Part of <span>the <a href="https://www.ufl.edu/" title="Located at Gainesville, Florida, USA">University of Florida</a></span></em></div></div><p>I worked full-time as the lead software developer on the Phyloreferencing project. Our goal is to build an ontology of definitions for groups of related biological organisms, as well as the software and ontological infrastructure needed to create, edit, organize and test these definitions. We are building a demonstration website that will allow users to resolve these definitions on any evolutionary hypothesis. I built these tools using JavaScript in Node.js, Vue CLI, Java and Python.</p><p><ul class="uncompressed roleProducts"><li id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project"><a class="section-link" href="#Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project">§</a><strong>Project:</strong> <a href="/projects#The-Phyloreferencing-Project">The Phyloreferencing Project</a> (<span title="January 2, 2018, 3 years, 3 months and 1 day ago">Jan 2018</span> <span class="no-title-dash" title="1 year, 9 months and 9 days">to</span> <span title="October 11, 2019, 1 year, 5 months and 21 days ago">Oct 2019</span>): <!-- -->The Phyloreferencing project aims to digitize definitions of clades (groups of related taxa that are a key concept in evolutionary biology) as semantically rich OWL ontologies, which are both human-readable and computer-interpretable.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project-Phyx.js" href="https://github.com/phyloref/phyx.js">Phyx.js</a> (<!-- -->from <span title="January 18, 2019, 2 years, 2 months and 14 days ago">Jan 2019</span>)<!-- -->: <!-- -->A software library for reading and validating Phyx files for storing clade definitions, which can convert them into OWL ontologies for publication and reasoning.</li><li><strong>Software</strong>: <a id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project-JPhyloRef" href="https://github.com/phyloref/jphyloref">JPhyloRef</a> (<span title="September 25, 2017, 3 years, 6 months and 7 days ago">Sep 2017</span> <span class="no-title-dash" title="2 years, 7 months and 23 days">to</span> <span title="May 18, 2020, 10 months and 14 days ago">May 2020</span>)<!-- -->: <!-- -->A Java-based command line tool for reasoning over and testing ontologies that include phyloreferences.</li><li><strong>Software</strong>: <a id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project-Klados" href="https://github.com/phyloref/klados">Klados</a> (<span title="October 12, 2017, 3 years, 5 months and 20 days ago">Oct 2017</span> <span class="no-title-dash" title="2 years, 9 months and 21 days">to</span> <span title="August 2, 2020, 8 months and 1 day ago">Aug 2020</span>)<!-- -->: <!-- -->A single-page application for curating, testing and resolving phyloreferences.</li><li><strong>Software</strong>: <a id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project-Open-Tree-Resolver" href="https://github.com/phyloref/open-tree-resolver">Open Tree Resolver</a> (<span title="February 22, 2019, 2 years, 1 month and 10 days ago">Feb 2019</span> <span class="no-title-dash" title="1 year, 6 months and 16 days">to</span> <span title="September 7, 2020, 6 months and 25 days ago">Sep 2020</span>)<!-- -->: <!-- -->A single-page application for testing phyloreference resolution on the Open Tree of Life.</li><li><strong>Software</strong>: <a id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project-Phyloref-Ontology" href="https://github.com/phyloref/phyloref-ontology">Phyloref Ontology</a> (<span title="June 12, 2013, 7 years, 9 months and 20 days ago">Jun 2013</span> <span class="no-title-dash" title="7 years, 5 months and 7 days">to</span> <span title="November 19, 2020, 4 months and 13 days ago">Nov 2020</span>)<!-- -->: <!-- -->An ontology containing terms used to concisely and precisely define phyloreferences.</li><li><strong>Software</strong>: <a id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-The-Phyloreferencing-Project-Clade-Ontology" href="https://github.com/phyloref/clade-ontology">Clade Ontology</a> (<span title="June 20, 2017, 3 years, 9 months and 12 days ago">Jun 2017</span> <span class="no-title-dash" title="2 years, 11 months and 2 days">to</span> <span title="May 22, 2020, 10 months and 10 days ago">May 2020</span>)<!-- -->: <!-- -->An ontology containing phyloreferences translated from published clade definitions.</li></ul><li><a class="section-link" href="#Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-https:doi.org10.3897BDJ.7.e33303">§</a><strong id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-https:doi.org10.3897BDJ.7.e33303">Citation</strong>: <!-- -->Brian Stucky<!-- -->, <a href="http://orcid.org/0000-0002-8688-6599">James Balhoff</a>, <!-- -->Narayani Barve<!-- -->, <a href="http://orcid.org/0000-0002-4852-2567">Vijay Barve</a>, <!-- -->Laura Brenskelle<!-- -->, <!-- -->Matthew Brush<!-- -->, <!-- -->Gregory Dahlem<!-- -->, <!-- -->James Gilbert<!-- -->, <a href="http://orcid.org/0000-0002-3724-4610">Akito Kawahara</a>, <!-- -->Oliver Keller<!-- -->, <!-- -->Andrea Lucky<!-- -->, <!-- -->Peter Mayhew<!-- -->, <a href="http://orcid.org/0000-0002-2339-655X">David Plotkin</a>, <a href="http://orcid.org/0000-0001-5354-6048">Katja Seltmann</a>, <!-- -->Elijah Talamas<!-- -->, <strong><em>Gaurav Vaidya</em></strong>, <a href="http://orcid.org/0000-0001-8815-0078">Ramona Walls</a>, <a href="http://orcid.org/0000-0002-5640-5491">Matt Yoder</a>, <a href="http://orcid.org/0000-0003-4389-4270">Guanyang Zhang</a>, <!-- -->Rob Guralnick<!-- --> <!-- -->(<!-- -->March 13, 2019<!-- -->)<!-- --> <a href="https://doi.org/10.3897/BDJ.7.e33303">Developing a vocabulary and ontology for modeling insect natural history data: example data, use cases, and competency questions</a>.<!-- --> <em>Biodiversity Data Journal</em> <strong>7</strong>.</li><li><a class="section-link" href="#Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-https:doi.org10.3897biss.4.59088">§</a><strong id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-https:doi.org10.3897biss.4.59088">Citation</strong>: <strong><em>Gaurav Vaidya</em></strong>, <a href="http://orcid.org/0000-0001-9107-0714">Hilmar Lapp</a>, <!-- -->Nico Cellinese<!-- --> <!-- -->(<!-- -->October 2, 2020<!-- -->)<!-- --> <a href="https://doi.org/10.3897/biss.4.59088">Enabling Machines to Integrate Biodiversity Data with Evolutionary Knowledge</a>.<!-- --> <em>Biodiversity Information Science and Standards</em> <strong>4</strong>.</li><li><a class="section-link" href="#Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-https:doi.org10.3897biss.2.25776">§</a><strong id="Postdoctoral-associate-the-Florida-Museum-of-Natural-History-at-Gainesville-Florida-USA-https:doi.org10.3897biss.2.25776">Citation</strong>: <strong><em>Gaurav Vaidya</em></strong>, <a href="http://orcid.org/0000-0003-4389-4270">Guanyang Zhang</a>, <a href="http://orcid.org/0000-0001-9107-0714">Hilmar Lapp</a>, <!-- -->Nico Cellinese<!-- --> <!-- -->(<!-- -->May 21, 2018<!-- -->)<!-- --> <a href="https://doi.org/10.3897/biss.2.25776">All the Clades in the World: Building a Semantically-Rich and Testable Ontology of Phylogenetic Clade Definitions</a>.<!-- --> <em>Biodiversity Information Science and Standards</em> <strong>2</strong>:e25776<!-- -->.</li></ul></p></div><div id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA">§</a><span title="August 22, 2011, 9 years, 7 months and 10 days ago">Aug 2011</span> <span class="no-title-dash" title="6 years, 3 months and 28 days">to</span> <span title="December 20, 2017, 3 years, 3 months and 12 days ago">Dec 2017</span> <!-- -->•<!-- --> <strong>PhD in Ecology and Evolutionary Biology</strong> <!-- -->at<!-- --> <span><a href="https://ebio.colorado.edu/" title="Located at Boulder, Colorado, USA">Ecology and Evolutionary Biology (EBIO)</a></span></div><div><div><em>Part of <span>the <a href="https://www.colorado.edu/" title="Located at Boulder, Colorado, USA">University of Colorado Boulder</a></span></em></div></div><p>I went into graduate school with two goals: (1) to become a scientific software developer, and (2) to spend a period of time doing a focussed study on the informatics of taxonomic names. Taxonomic names are one of the oldest information management schemes in science, and have had to change dramatically over the last 285 years as biologists&#x27; understanding of species and evolution has changed. I wondered what we could learn about this old-yet-new information management system. Apart from the projects listed here that were directly relevant to my PhD, I also spent my time in graduate school learning about several other tools and techniques in a variety of other roles, listed elsewhere.</p><p><ul class="uncompressed roleProducts"><li id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation"><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation">§</a><strong>Project:</strong> <a href="/projects#PhD-dissertation">PhD dissertation</a> (<span title="August 22, 2011, 9 years, 7 months and 10 days ago">Aug 2011</span> <span class="no-title-dash" title="9 years, 3 months and 28 days">to</span> <span title="December 20, 2020, 3 months and 12 days ago">Dec 2020</span>): <!-- -->In my PhD dissertation, I developed methods for quantifying the rate of change with taxonomic checklists, and the effect such changes can have on the interpretation of biodiversity data.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-SciNames" href="https://github.com/gaurav/scinames">SciNames</a> (<span title="January 1, 2017, 4 years, 3 months and 1 day ago">Jan 2017</span> <span class="no-title-dash" title="2 years, 4 months and 25 days">to</span> <span title="May 26, 2019, 1 year, 10 months and 6 days ago">May 2019</span>)<!-- -->: <!-- -->A graphical user interface (GUI) for processing changes among taxonomic checklists. SciNames imports multiple taxonomic checklists into a single XML file, and then generates lists of changes between those checklists. Users can then use the GUI to annotate these changes, recording why each change had occurred and what sort of change it was. SciNames can then calculate some statistics on the changes.</li><li><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-https:doi.org10.31237osf.ion2kh5">§</a><strong id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-https:doi.org10.31237osf.ion2kh5">Citation</strong>: <strong><em>Gaurav Vaidya</em></strong> <!-- -->(<!-- -->January 19, 2018<!-- -->)<!-- --> <a href="https://doi.org/10.31237/osf.io/n2kh5">Taxonomic Checklists as Biodiversity Data: How Series of Checklists can Provide Information on Synonymy, Circumscription Change and Taxonomic Discovery</a>.<!-- --> <em>Ph.D. Dissertation</em> <strong></strong>.</li><li><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-https:doi.org10.1371journal.pone.0195736">§</a><strong id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-https:doi.org10.1371journal.pone.0195736">Citation</strong>: <strong><em>Gaurav Vaidya</em></strong>, <a href="https://avibase.bsc-eoc.org/about.jsp">Denis Lepage</a>, <a href="https://sites.google.com/site/robgur/">Robert Guralnick</a> <!-- -->(<!-- -->April 19, 2018<!-- -->)<!-- --> <a href="https://doi.org/10.1371/journal.pone.0195736">The tempo and mode of the taxonomic correction process: How taxonomists have corrected and recorrected North American bird species over the last 127 years</a>.<!-- --> <em>PLOS ONE</em> <strong>13</strong>(4)<!-- -->:e0195736<!-- -->.</li><li><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-https:doi.org10.3897zookeys.420.7089">§</a><strong id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-PhD-dissertation-https:doi.org10.3897zookeys.420.7089">Citation</strong>: <a href="https://avibase.bsc-eoc.org/about.jsp">Denis Lepage</a>, <strong><em>Gaurav Vaidya</em></strong>, <a href="https://sites.google.com/site/robgur/">Robert Guralnick</a> <!-- -->(<!-- -->June 25, 2014<!-- -->)<!-- --> <a href="https://doi.org/10.3897/zookeys.420.7089">Avibase – a database system for managing and organizing taxonomic concepts</a>.<!-- --> <em>ZooKeys</em> <strong>420</strong>:117–135<!-- -->.</li></ul><li id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-Small-projects-during-my-PhD"><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-Small-projects-during-my-PhD">§</a><strong>Project:</strong> <a href="/projects#Small-projects-during-my-PhD">Small projects during my PhD</a> (<span title="January 1, 2011, 10 years, 3 months and 1 day ago">Jan 2011</span> <span class="no-title-dash" title="6 years, zero months and zero days">to</span> <span title="January 1, 2017, 4 years, 3 months and 1 day ago">Jan 2017</span>): <!-- -->Some small software projects I worked on during my PhD.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-Small-projects-during-my-PhD-BibURI" href="https://github.com/gaurav/biburi">BibURI</a> (<span title="October 1, 2013, 7 years, 6 months and 1 day ago">Oct 2013</span> <span class="no-title-dash" title="3 months and 26 days">to</span> <span title="January 27, 2014, 7 years, 2 months and 5 days ago">Jan 2014</span>)<!-- -->: <!-- -->A Ruby Gem to extract BibTeX information from a number of bibliographic resources, such as DOIs and COinS webpages. I wrote this at the first TaxonWorks hackathon.</li></ul><li><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-https:doi.org10.11861471-2105-14-158">§</a><strong id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-https:doi.org10.11861471-2105-14-158">Citation</strong>: <!-- -->Arlin Stoltzfus<!-- -->, <!-- -->Hilmar Lapp<!-- -->, <!-- -->Naim Matasci<!-- -->, <!-- -->Helena Deus<!-- -->, <!-- -->Brian Sidlauskas<!-- -->, <!-- -->Christian M Zmasek<!-- -->, <strong><em>Gaurav Vaidya</em></strong>, <!-- -->Enrico Pontelli<!-- -->, <!-- -->Karen Cranston<!-- -->, <!-- -->Rutger Vos<!-- -->, <!-- -->Campbell O Webb<!-- -->, <!-- -->Luke J Harmon<!-- -->, <!-- -->Megan Pirrung<!-- -->, <!-- -->Brian O&#x27;Meara<!-- -->, <!-- -->Matthew W Pennell<!-- -->, <!-- -->Siavash Mirarab<!-- -->, <!-- -->Michael S Rosenberg<!-- -->, <!-- -->James P Balhoff<!-- -->, <!-- -->Holly M Bik<!-- -->, <!-- -->Tracy A Heath<!-- -->, <!-- -->Peter E Midford<!-- -->, <!-- -->Joseph W Brown<!-- -->, <!-- -->Emily Jane McTavish<!-- -->, <!-- -->Jeet Sukumaran<!-- -->, <!-- -->Mark Westneat<!-- -->, <!-- -->Michael E Alfaro<!-- -->, <!-- -->Aaron Steele<!-- -->, <!-- -->Greg Jordan<!-- --> <!-- -->(<!-- -->May 13, 2013<!-- -->)<!-- --> <a href="https://doi.org/10.1186/1471-2105-14-158">Phylotastic! Making tree-of-life knowledge accessible, reusable and convenient</a>.<!-- --> <em>BMC Bioinformatics</em> <strong>14</strong>(1)<!-- -->:158<!-- -->.</li><li><a class="section-link" href="#PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-https:doi.org10.3897zookeys.209.3247">§</a><strong id="PhD-in-Ecology-and-Evolutionary-Biology-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-https:doi.org10.3897zookeys.209.3247">Citation</strong>: <!-- -->Andrea Thomer<!-- -->, <strong><em>Gaurav Vaidya</em></strong>, <!-- -->Robert Guralnick<!-- -->, <!-- -->David Bloom<!-- -->, <!-- -->Laura Russell<!-- --> <!-- -->(<!-- -->July 20, 2012<!-- -->)<!-- --> <a href="https://doi.org/10.3897/zookeys.209.3247">From documents to datasets: A MediaWiki-based method of annotating and extracting species observations in century-old field notebooks</a>.<!-- --> <em>ZooKeys</em> <strong>209</strong>:235-253<!-- -->.</li></ul></p></div><div id="Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA">§</a><span title="January 1, 2015, 6 years, 3 months and 1 day ago">Jan 2015</span> <span class="no-title-dash" title="1 year, 4 months and zero days">to</span> <span title="May 1, 2016, 4 years, 11 months and 1 day ago">May 2016</span> <!-- -->•<!-- --> <strong>Graduate teaching assistant</strong> <!-- -->at<!-- --> <span><a href="https://ebio.colorado.edu/" title="Located at Boulder, Colorado, USA">Ecology and Evolutionary Biology (EBIO)</a></span></div><div><div><em>Part of <span>the <a href="https://www.colorado.edu/" title="Located at Boulder, Colorado, USA">University of Colorado Boulder</a></span></em></div></div><p>While a graduate student, I taught three Evolutionary Biology and General Biology labs, in which I led classes of up to eighteen undergraduate students through hands-on exercises to build their understanding of biology and evolution. CU Boulder asks students to evaluate their instructors on a scale from 1 (worst) to 6 (best); my reviews increased from 4.8-5.5 in my first semester to 5.3-5.6 in my last semester.</p><p><ul class="uncompressed roleProducts"><li id="Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-General-Biology-Labs-2-(EBIO-1240)"><a class="section-link" href="#Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-General-Biology-Labs-2-(EBIO-1240)">§</a><strong>Project:</strong> <a href="/projects#General-Biology-Labs-2-(EBIO-1240)">General Biology Labs 2 (EBIO 1240)</a> (<span title="January 1, 2016, 5 years, 3 months and 1 day ago">Jan 2016</span> <span class="no-title-dash" title="4 months and zero days">to</span> <span title="May 1, 2016, 4 years, 11 months and 1 day ago">May 2016</span>): <!-- -->I led three lab sections of eighteen students each, taking my students on a tour of the tree of life while reinforcing concepts in evolutionary biology, ecology, anatomy and physiology.</li><li id="Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-General-Biology-Labs-1-(EBIO-1230)"><a class="section-link" href="#Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-General-Biology-Labs-1-(EBIO-1230)">§</a><strong>Project:</strong> <a href="/projects#General-Biology-Labs-1-(EBIO-1230)">General Biology Labs 1 (EBIO 1230)</a> (<span title="August 1, 2015, 5 years, 8 months and 1 day ago">Aug 2015</span> <span class="no-title-dash" title="4 months and zero days">to</span> <span title="December 1, 2015, 5 years, 4 months and 1 day ago">Dec 2015</span>): <!-- -->I led three lab sections of eighteen students each, teaching the philosophical underpinings of science through hands-on experiments in cellular and molecular biology.</li><li id="Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-Evolutionary-Biology-(EBIO-3080)"><a class="section-link" href="#Graduate-teaching-assistant-Ecology-and-Evolutionary-Biology-(EBIO)-at-Boulder-Colorado-USA-Evolutionary-Biology-(EBIO-3080)">§</a><strong>Project:</strong> <a href="/projects#Evolutionary-Biology-(EBIO-3080)">Evolutionary Biology (EBIO 3080)</a> (<span title="January 1, 2015, 6 years, 3 months and 1 day ago">Jan 2015</span> <span class="no-title-dash" title="4 months and zero days">to</span> <span title="May 1, 2015, 5 years, 11 months and 1 day ago">May 2015</span>): <!-- -->I led three lab sections of twenty students each, teaching evolutionary biology through R-based statistics and modeling labs, measurements and phylogenetics.</li></ul></p></div><div id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA">§</a><span title="August 1, 2011, 9 years, 8 months and 1 day ago">Aug 2011</span> <span class="no-title-dash" title="4 years, 4 months and zero days">to</span> <span title="December 1, 2015, 5 years, 4 months and 1 day ago">Dec 2015</span> <!-- -->•<!-- --> <strong>Graduate research assistant</strong> <!-- -->at<!-- --> <span>the <a href="https://ebio.colorado.edu/" title="Located at Boulder, Colorado, USA">University of Colorado Museum of Natural History</a></span></div><div><div><em>Part of <span>the <a href="https://www.colorado.edu/cumuseum/" title="Located at Boulder, Colorado, USA">University of Colorado Boulder</a></span></em></div></div><p>During the first four years of my PhD, I worked on the Map of Life and VertNet projects, where I developed a web application for synthesizing and managing vernacular names extracted from multiple sources, a web API for efficiently searching a large database of taxonomic names, and a Python tool to identify errors in species names (now deprecated).</p><p><ul class="uncompressed roleProducts"><li id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Art-of-Life"><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Art-of-Life">§</a><strong>Project:</strong> <a href="/projects#Art-of-Life">Art of Life</a> (<span title="May 1, 2012, 8 years, 11 months and 1 day ago">May 2012</span> <span class="no-title-dash" title="2 years, 11 months and zero days">to</span> <span title="April 1, 2015, 6 years, zero months and 1 day ago">Apr 2015</span>): <!-- -->A project organized by the Biodiversity Heritage Library (BHL) to identical and annotate hundreds of thousands of illustrations from the documents in this digital library.</li><ul class="uncompressed"><li><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Art-of-Life-Art-of-Life-Schema">§</a><strong>Link</strong>: <a id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Art-of-Life-Art-of-Life-Schema" href="https://docs.google.com/document/d/1jjHMjWWnsErTLtms5MVuBg2wE6wtOuGmA0-UVSk5_Ts/edit">Art of Life Schema</a> (<span title="April 30, 2015 to April 30, 2015, zero days ago">Apr 2015</span>)<!-- -->: <!-- -->I worked with my PhD advisor and some librarians at the BHL to develop a data schema for annotating biological illustrations in a way that would make them useful for biodiversity researchers.</li></ul><li id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Map-of-Life"><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Map-of-Life">§</a><strong>Project:</strong> <a href="/projects#Map-of-Life">Map of Life</a> (<span title="January 1, 2011, 10 years, 3 months and 1 day ago">Jan 2011</span> <span class="no-title-dash" title="4 years, zero months and zero days">to</span> <span title="January 1, 2015, 6 years, 3 months and 1 day ago">Jan 2015</span>): <!-- -->An NSF-funded project to synthesize different kinds of biodiversity data — from occurrences to rangemaps — into a single, easy-to-use tool. I worked on the Map of Life project, usually during summer holidays, under the supervision of my PhD advisor, Rob Guralnick.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Map-of-Life-Vernacular-Names" href="https://github.com/MapofLife/vernacular-names">Vernacular Names</a> (<span title="February 28, 2014, 7 years, 1 month and 4 days ago">Feb 2014</span> <span class="no-title-dash" title="1 year, 4 months and 26 days">to</span> <span title="July 24, 2015, 5 years, 8 months and 8 days ago">Jul 2015</span>)<!-- -->: <!-- -->A web application for managing the synthesis and verification of vernacular name information for Map of Life.</li><li><strong>Software</strong>: <a id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Map-of-Life-TaxRefine" href="https://github.com/gaurav/taxrefine">TaxRefine</a> (<span title="June 17, 2013, 7 years, 9 months and 15 days ago">Jun 2013</span> <span class="no-title-dash" title="11 months and 11 days">to</span> <span title="May 28, 2014, 6 years, 10 months and 4 days ago">May 2014</span>)<!-- -->: <!-- -->Provides an OpenRefine reconcilation service API for matching taxonomic names against several services.</li><li><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Map-of-Life-Validating-scientific-names-with-the-GBIF-Checklist-Bank">§</a><strong>Link</strong>: <a id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-Map-of-Life-Validating-scientific-names-with-the-GBIF-Checklist-Bank" href="https://blog.vertnet.org/post/56169017224/taxonomic-validation-vaidya">Validating scientific names with the GBIF Checklist Bank</a> (<span title="July 22, 2013 to July 22, 2013, zero days ago">Jul 2013</span>)<!-- -->: <!-- -->A blog post describing TaxRefine.</li></ul><li id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-The-Junius-Henderson-Field-Notebook-Project"><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-The-Junius-Henderson-Field-Notebook-Project">§</a><strong>Project:</strong> <a href="/projects#The-Junius-Henderson-Field-Notebook-Project">The Junius Henderson Field Notebook Project</a> (<span title="November 1, 2011, 9 years, 5 months and 1 day ago">Nov 2011</span> <span class="no-title-dash" title="8 months and 19 days">to</span> <span title="July 20, 2012, 8 years, 8 months and 12 days ago">Jul 2012</span>): <!-- -->This project began as an idea that my collaborators (Andrea Thomer and Rob Guralnick) discussed regarding the annotation of historical field notebooks to extract the biodiversity data found in them. I suggested that we use Wikisource, a part of Wikipedia, to crowdsource the annotation process and to develop the software tools needed to carry out such annotations of Wikisource content. We published our findings as a Darwin Core Archive and in the journal ZooKeys in 2012.</li><ul class="uncompressed"><li><a class="section-link" href="#Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-The-Junius-Henderson-Field-Notebook-Project-https:doi.org10.3897zookeys.209.3247">§</a><strong id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-The-Junius-Henderson-Field-Notebook-Project-https:doi.org10.3897zookeys.209.3247">Citation</strong>: <!-- -->Andrea Thomer<!-- -->, <strong><em>Gaurav Vaidya</em></strong>, <!-- -->Robert Guralnick<!-- -->, <!-- -->David Bloom<!-- -->, <!-- -->Laura Russell<!-- --> <!-- -->(<!-- -->July 20, 2012<!-- -->)<!-- --> <a href="https://doi.org/10.3897/zookeys.209.3247">From documents to datasets: A MediaWiki-based method of annotating and extracting species observations in century-old field notebooks</a>.<!-- --> <em>ZooKeys</em> <strong>209</strong>:235-253<!-- -->.</li><li><strong>Software</strong>: <a id="Graduate-research-assistant-the-University-of-Colorado-Museum-of-Natural-History-at-Boulder-Colorado-USA-The-Junius-Henderson-Field-Notebook-Project-The-Junius-Henderson-Field-Notebook-Project-source-code" href="https://github.com/gaurav/henderson">The Junius Henderson Field Notebook Project source code</a> (<span title="January 19, 2012, 9 years, 2 months and 13 days ago">Jan 2012</span> <span class="no-title-dash" title="2 months and 12 days">to</span> <span title="March 31, 2012, 9 years, zero months and 1 days ago">Mar 2012</span>)<!-- -->: <!-- -->I wrote a series of scripts for extracting biodiversity data annotated on WikiSource into the Darwin Core biodiversity data sharing format.</li></ul></ul></p></div><div id="Student-developer-DBpedia-at-Boulder-Colorado-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Student-developer-DBpedia-at-Boulder-Colorado-USA">§</a><span title="May 1, 2014, 6 years, 11 months and 1 day ago">May 2014</span> <span class="no-title-dash" title="3 months and zero days">to</span> <span title="August 1, 2014, 6 years, 8 months and 1 day ago">Aug 2014</span> <!-- -->•<!-- --> <strong>Student developer</strong> <!-- -->at<!-- --> <span><a href="https://dbpedia.org/" title="Located at Boulder, Colorado, USA">DBpedia</a></span></div><div><div><em>Funded by <span><a href="https://summerofcode.withgoogle.com/">Google Summer of Code</a></span><span> <!-- -->(part of <span><a href="https://www.google.com/">Google</a></span>)</span></em></div></div><p>I extended DBpedia&#x27;s fact extraction software to support extracting facts in RDF from the Wikimedia Commons, an online repository that then contained around 25 million media files across a number of formats and licenses.</p><p><ul class="uncompressed roleProducts"><li id="Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code"><a class="section-link" href="#Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code">§</a><strong>Project:</strong> <a href="/projects#DBpedia-Commons-at-Google-Summer-of-Code">DBpedia Commons at Google Summer of Code</a> (<span title="May 19, 2014, 6 years, 10 months and 13 days ago">May 2014</span> <span class="no-title-dash" title="3 months and 3 days">to</span> <span title="August 22, 2014, 6 years, 7 months and 10 days ago">Aug 2014</span>): <!-- -->For a Google Summer of Code project in summer 2014, I worked on extending the DBpedia Extraction Framework to be able to extract information from the Wikimedia Commons database dumps and make them available as RDF.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-DBpedia-Extraction-Framework" href="https://github.com/dbpedia/extraction-framework">DBpedia Extraction Framework</a> (<span title="May 31, 2014, 6 years, 10 months and 1 days ago">May 2014</span> <span class="no-title-dash" title="2 months and 18 days">to</span> <span title="August 18, 2014, 6 years, 7 months and 14 days ago">Aug 2014</span>)<!-- -->: <!-- -->I made some improvements to the DBpedia Extraction Framework to allow them to extract information from the Wikimedia Commons.</li><li><a class="section-link" href="#Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-https:doi.org10.1007978-3-319-25010-6_17">§</a><strong id="Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-https:doi.org10.1007978-3-319-25010-6_17">Citation</strong>: <strong><em>Gaurav Vaidya</em></strong>, <!-- -->Dimitris Kontokostas<!-- -->, <!-- -->Magnus Knuth<!-- -->, <!-- -->Jens Lehmann<!-- -->, <!-- -->Sebastian Hellmann<!-- --> <!-- -->(<!-- -->October 24, 2015<!-- -->)<!-- --> <a href="https://doi.org/10.1007/978-3-319-25010-6_17">DBpedia Commons: Structured Multimedia Metadata from the Wikimedia Commons</a>.<!-- --> <em>The Semantic Web - ISWC 2015: Lecture Notes in Computer Science</em> <strong>9367</strong>:281-289<!-- -->.</li><li><a class="section-link" href="#Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-Project-description-on-Google-Summer-of-Code&#x27;s-website">§</a><strong>Link</strong>: <a id="Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-Project-description-on-Google-Summer-of-Code&#x27;s-website" href="https://www.google-melange.com/archive/gsoc/2014/orgs/dbpediaspotlight/projects/gaurav.html">Project description on Google Summer of Code&#x27;s website</a></li><li><a class="section-link" href="#Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-Project-report-on-the-Wikimedia-Commons">§</a><strong>Link</strong>: <a id="Student-developer-DBpedia-at-Boulder-Colorado-USA-DBpedia-Commons-at-Google-Summer-of-Code-Project-report-on-the-Wikimedia-Commons" href="https://commons.wikimedia.org/wiki/User:Gaurav/DBpedia">Project report on the Wikimedia Commons</a></li></ul></ul></p></div><div id="Graduate-research-assistant-the-National-Evolutionary-Synthesis-Center-(NESCent)-at-Durham-North-Carolina-USA" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Graduate-research-assistant-the-National-Evolutionary-Synthesis-Center-(NESCent)-at-Durham-North-Carolina-USA">§</a><span title="January 1, 2013, 8 years, 3 months and 1 day ago">Jan 2013</span> <span class="no-title-dash" title="4 months and zero days">to</span> <span title="May 1, 2013, 7 years, 11 months and 1 day ago">May 2013</span> <!-- -->•<!-- --> <strong>Graduate research assistant</strong> <!-- -->at<!-- --> <span>the <a href="https://www.nescent.org/" title="Located at Durham, North Carolina, USA">National Evolutionary Synthesis Center (NESCent)</a></span></div><div></div><p>This fellowship allowed me to work exclusively on my PhD for one semester with a mentor at the National Evolutionary Synthesis Center (NESCent) in Durham, North Carolina. My mentor at NESCent, Hilmar Lapp, would later hire me to work on the Phyloreferencing project.</p><p><ul class="uncompressed roleProducts"><li><a class="section-link" href="#Graduate-research-assistant-the-National-Evolutionary-Synthesis-Center-(NESCent)-at-Durham-North-Carolina-USA-Description-of-my-project-on-NESCent&#x27;s-website">§</a><strong>Link</strong>: <a id="Graduate-research-assistant-the-National-Evolutionary-Synthesis-Center-(NESCent)-at-Durham-North-Carolina-USA-Description-of-my-project-on-NESCent&#x27;s-website" href="https://www.nescent.org/science/awards_summary.php-id=374.html">Description of my project on NESCent&#x27;s website</a></li></ul></p></div><div id="Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore">§</a><span title="November 1, 2007, 13 years, 5 months and 1 day ago">Nov 2007</span> <span class="no-title-dash" title="3 years, 3 months and 27 days">to</span> <span title="February 28, 2011, 10 years, 1 month and 4 days ago">Feb 2011</span> <!-- -->•<!-- --> <strong>Software architect</strong> <!-- -->at<!-- --> <span><a href="http://www.paperterminal.org/" title="Located at Singapore">Paper Terminal Pte Ltd</a></span></div><div></div><p>I was the primary software developer, responsible for developing new web applications from prototypes to final deployment, including OCR Terminal, my company&#x27;s flagship product. I also managed our computer systems on both on-site and Amazon EC2 cloud plaform across multiple operating systems.</p><p><ul class="uncompressed roleProducts"><li id="Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore-OCR-Terminal"><a class="section-link" href="#Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore-OCR-Terminal">§</a><strong>Project:</strong> <a href="/projects#OCR-Terminal">OCR Terminal</a> (<span title="January 1, 2008, 13 years, 3 months and 1 day ago">Jan 2008</span> <span class="no-title-dash" title="3 years, zero months and zero days">to</span> <span title="January 1, 2011, 10 years, 3 months and 1 day ago">Jan 2011</span>): <!-- -->A start-up that focussed on providing services around optical character recognition (OCR) technology.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore-OCR-Terminal-OCR-Terminal" href="http://www.ocrterminal.com/">OCR Terminal</a> (<span title="February 1, 2008, 13 years, 2 months and 1 day ago">Feb 2008</span> <span class="no-title-dash" title="2 years, 10 months and 23 days">to</span> <span title="December 24, 2010, 10 years, 3 months and 8 days ago">Dec 2010</span>)<!-- -->: <p>OCR Terminal was an online optical character recognition (OCR) service: it read text from uploaded images and provided the image files in an editable format such as Microsoft Word, Adobe PDF or plain text. Between mid-2008 and 2011, tens of thousands of user accounts have been created and over 100,000 documents had been processed on this website. Apart from the website itself, the service featured a simple API which can be used to submit documents for processing programmatically.</p><p>I was lead developer of OCR Terminal right from project inception. I have written all of OCR Terminal&#x27;s underlying code, first as a Perl/CGI application, and later as a Perl/Catalyst application. I am particularly proud of designing the public API, which is used by our own desktop client, several in-house tools, and several clients of ours who use it for both bulk processing and as a backend for their own software.</p><p>I also manage OCR Terminal&#x27;s main server administrator, responsible for maintaining all the servers and backend components. I was able to learn about server monitoring with tools such as Munin. Since early 2009, OCR Terminal has been hosted on the Amazon EC2 processing cloud, giving me experience with setting up, bundling and managing EC2 instances.&lt;/p&gt;</p></li></ul><li><a class="section-link" href="#Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore-Description-of-OCR-Terminal-on-ABBYY&#x27;s-website">§</a><strong>Link</strong>: <a id="Software-architect-Paper-Terminal-Pte-Ltd-at-Singapore-Description-of-OCR-Terminal-on-ABBYY&#x27;s-website" href="https://www.abbyy.com/customer-stories/abbyy-recognition-server-proves-to-be-credible-infrastructure-for-online-services-ocr-terminal/">Description of OCR Terminal on ABBYY&#x27;s website</a></li></ul></p></div><div id="Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore">§</a><span title="August 22, 2006, 14 years, 7 months and 10 days ago">Aug 2006</span> <span class="no-title-dash" title="10 months and zero days">to</span> <span title="June 22, 2007, 13 years, 9 months and 10 days ago">Jun 2007</span> <!-- -->•<!-- --> <strong>Lab officer</strong> <!-- -->at<!-- --> <span>the <a href="http://evolution.science.nus.edu.sg" title="Located at Singapore">Evolutionary Biology Laboratory</a></span></div><div><div><em>Part of <span>the <a href="http://www.nus.edu.sg/" title="Located at Singapore">National University of Singapore</a></span></em></div></div><p>I helped manage computer-related infrastructure, from sending computers for servicing to installing scientific software on both local hardware and remote computing clusters. I also finished work on several scientific tools, which I have documented under my educational history below.</p><p><ul class="uncompressed roleProducts"><li id="Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-SequenceMatrix"><a class="section-link" href="#Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-SequenceMatrix">§</a><strong>Project:</strong> <a href="/projects#SequenceMatrix">SequenceMatrix</a> (<span title="August 1, 2006, 14 years, 8 months and 1 day ago">Aug 2006</span> <span class="no-title-dash" title="4 years, 3 months and 12 days">to</span> <span title="November 13, 2010, 10 years, 4 months and 19 days ago">Nov 2010</span>): <!-- -->Genetic analysis softwares at this time were designed to compare a single genetic or protein sequence between different taxa to perform comparative analyses. In the Evolutionary Biology lab, we would often perform multi-gene analyses, requiring aligned genetic sequences to be concatenated together. Modifying such datasets after concatenation could be problematic, especially if one of the constituent sequences turned out to be contamination. Sequence Matrix was intended to simplify the process of contatenation such that preserved gene boundaries could be unconcatenated if necessary, and included some tools for detecting and removing contamination from multi-gene, multi-taxon datasets.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-SequenceMatrix-SequenceMatrix" href="http://gaurav.github.io/taxondna/">SequenceMatrix</a> (<span title="August 28, 2006, 14 years, 7 months and 4 days ago">Aug 2006</span> <span class="no-title-dash" title="8 years, 9 months and 25 days">to</span> <span title="June 22, 2015, 5 years, 9 months and 10 days ago">Jun 2015</span>)<!-- -->: <!-- -->Sequence Matrix facilitates the assembly of phylogenetic data matrices with multiple genes. Files for individual genes are dragged and dropped into a window and the sequences are concatenated. A table provides an overview over how much sequence information is available for the different genes and species. The user can request Sequence Matrix generate a wide variety of character and taxon sets (e.g. a taxon set with all species that have more than a specified number of genes or basepairs). The concatenated sequences can be exported in NEXUS or TNT format. Individual sequences can be excluded from being exported.</li><li><a class="section-link" href="#Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-SequenceMatrix-https:doi.org10.1111j.1096-0031.2010.00329.x">§</a><strong id="Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-SequenceMatrix-https:doi.org10.1111j.1096-0031.2010.00329.x">Citation</strong>: <strong><em>Gaurav Vaidya</em></strong>, <a href="http://lohman.ccny.cuny.edu/">David J. Lohman</a>, <a href="https://www.dbs.nus.edu.sg/staffs/rudolf-meier/">Rudolf Meier</a> <!-- -->(<!-- -->March 8, 2011<!-- -->)<!-- --> <a href="https://doi.org/10.1111/j.1096-0031.2010.00329.x">SequenceMatrix: concatenation software for the fast assembly of multi‐gene datasets with character set and codon information</a>.<!-- --> <em>Cladistics</em> <strong>27</strong>(2)<!-- -->:171–180<!-- -->.</li></ul><li><a class="section-link" href="#Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-https:doi.org10.1016j.ympev.2011.11.034">§</a><strong id="Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-https:doi.org10.1016j.ympev.2011.11.034">Citation</strong>: <!-- -->Shiyang Kwong<!-- -->, <!-- -->Amrita Srivathsan<!-- -->, <strong><em>Gaurav Vaidya</em></strong>, <!-- -->Rudolf Meier<!-- --> <!-- -->(<!-- -->December 11, 2011<!-- -->)<!-- --> <a href="https://doi.org/10.1016/j.ympev.2011.11.034">Is the COI barcoding gene involved in speciation through intergenomic conflict?</a>.<!-- --> <em>Molecular Phylogenetics and Evolution</em> <strong>62</strong>(3)<!-- -->:1009-1012<!-- -->.</li><li><a class="section-link" href="#Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-My-description-on-our-lab-website">§</a><strong>Link</strong>: <a id="Lab-officer-the-Evolutionary-Biology-Laboratory-at-Singapore-My-description-on-our-lab-website" href="https://web.archive.org/web/20080521174221/http://evolution.science.nus.edu.sg/people/gaurav/">My description on our lab website</a></li></ul></p></div><div id="Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore" class="timeline-module--timelineEvent--1GFNQ"><div><a class="section-link" href="#Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore">§</a><span title="July 1, 2002, 18 years, 9 months and 1 day ago">Jul 2002</span> <span class="no-title-dash" title="3 years, 11 months and 29 days">to</span> <span title="June 30, 2006, 14 years, 9 months and 2 days ago">Jun 2006</span> <!-- -->•<!-- --> <strong>Bachelor of Science (with Merit) in Life Sciences with minors in Computational Science and Economics</strong> <!-- -->at<!-- --> <span><a href="https://www.dbs.nus.edu.sg/" title="Located at Singapore">Department of Biological Sciences</a></span></div><div><div><em>Part of <span>the <a href="http://www.nus.edu.sg/" title="Located at Singapore">National University of Singapore</a></span></em></div></div><p><ul class="uncompressed roleProducts"><li id="Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-Species-Identifier"><a class="section-link" href="#Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-Species-Identifier">§</a><strong>Project:</strong> <a href="/projects#Species-Identifier">Species Identifier</a> (<span title="January 1, 2003, 18 years, 3 months and 1 day ago">Jan 2003</span> <span class="no-title-dash" title="3 years, zero months and zero days">to</span> <span title="January 1, 2006, 15 years, 3 months and 1 day ago">Jan 2006</span>): <!-- -->This project came about in response to a need for a series of projects by the Evolutionary Biology lab to investigate whether genetic distance methods could be used to correctly identify the species that a genetic sequence had been sequenced from. The tool we built allowed us to compare several different approaches to investigate this approach, and then to replicate these analyses on larger datasets than we had originally envisioned.</li><ul class="uncompressed"><li><strong>Software</strong>: <a id="Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-Species-Identifier-Species-Identifier" href="http://taxondna.sf.net/">Species Identifier</a> (<span title="August 20, 2006, 14 years, 7 months and 12 days ago">Aug 2006</span> <span class="no-title-dash" title="8 years, 10 months and 2 days">to</span> <span title="June 22, 2015, 5 years, 9 months and 10 days ago">Jun 2015</span>)<!-- -->: <!-- -->Species Identifier provides a set of tools for exploring intra- and interspecific genetic distances, matching sequences, and clustering sequences based on pairwise distances. It helps determine whether two sequences are likely conspecific based on pairwise distances, and can calculate pairwise distances for large datasets. It was designed to provide the analyses presented in Meier et al, 2006 (Best match, Best close match and All Species Barcodes).</li><li><a class="section-link" href="#Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-Species-Identifier-https:doi.org10.108010635150600969864">§</a><strong id="Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-Species-Identifier-https:doi.org10.108010635150600969864">Citation</strong>: <a href="https://www.dbs.nus.edu.sg/staffs/rudolf-meier/">Rudolf Meier</a>, <a href="https://www.linkedin.com/in/shiyang-kwong-b490a356/">Kwong Shiyang</a>, <strong><em>Gaurav Vaidya</em></strong>, <a href="https://www.wikidata.org/wiki/Q21393244">Peter K. L. Ng</a> <!-- -->(<!-- -->October 1, 2006<!-- -->)<!-- --> <a href="https://doi.org/10.1080/10635150600969864">DNA Barcoding and Taxonomy in Diptera: A Tale of High Intraspecific Variability and Low Identification Success</a>.<!-- --> <em>Systematic Biology</em> <strong>55</strong>(5)<!-- -->:715–728<!-- -->.</li></ul><li><a class="section-link" href="#Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-https:doi.org10.1163ej.9789004148970.I-459.55">§</a><strong id="Bachelor-of-Science-(with-Merit)-in-Life-Sciences-with-minors-in-Computational-Science-and-Economics-Department-of-Biological-Sciences-at-Singapore-https:doi.org10.1163ej.9789004148970.I-459.55">Citation</strong>: <a href="http://orcid.org/0000-0003-4816-2909">Torsten Dikow</a>, <!-- -->Rudolf Meier<!-- -->, <strong><em>Gaurav G. Vaidya</em></strong>, <!-- -->Jason G. H. Londt<!-- --> <!-- -->(<!-- -->March 25, 2009<!-- -->)<!-- --> <a href="https://doi.org/10.1163/ej.9789004148970.I-459.55">Chapter Twelve. Biodiversity Research Based On Taxonomic Revisions — A Tale Of Unrealized Opportunities</a>.<!-- --> <em>Diptera Diversity: Status, Challenges and Tools</em> <strong></strong>:323-346<!-- -->.</li></ul></p></div></main><footer>© <!-- -->2021<!-- --> Gaurav Vaidya. Built with<!-- --> <a href="https://www.gatsbyjs.org">Gatsby</a>.<br/><small><a href="https://github.com/gaurav/gaurav.github.io">Source code available</a> under the GPL v3.0 open source license.<!-- --> <a href="https://github.com/gaurav/gaurav.github.io/tree/develop/jsonld">Raw data available</a> as RDF in JSON-LD and Turtle.</small></footer></div></div><div id="gatsby-announcer" style="position:absolute;top:0;width:1px;height:1px;padding:0;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;border:0" aria-live="assertive" aria-atomic="true"></div></div><script id="gatsby-script-loader">/*<![CDATA[*/window.pagePath="/timeline/";/*]]>*/</script><script id="gatsby-chunk-mapping">/*<![CDATA[*/window.___chunkMapping={"polyfill":["/polyfill-983cd605c9f54f544b64.js"],"app":["/app-1ff00ff6183944037c30.js"],"component---src-pages-404-js":["/component---src-pages-404-js-549ab0df15d4fb824a62.js"],"component---src-pages-index-js":["/component---src-pages-index-js-0dd775b731d16c5396ae.js"],"component---src-pages-projects-js":["/component---src-pages-projects-js-9c58104ffa88b5674817.js"],"component---src-pages-publications-js":["/component---src-pages-publications-js-c3563f9f63474d9dccc2.js"],"component---src-pages-tag-[id]-js":["/component---src-pages-tag-[id]-js-a5d0c969165c90378b13.js"],"component---src-pages-tags-js":["/component---src-pages-tags-js-85fc17bce5ede17a640c.js"],"component---src-pages-timeline-js":["/component---src-pages-timeline-js-dcc741760f588e8f0200.js"]};/*]]>*/</script><script src="/polyfill-983cd605c9f54f544b64.js" nomodule=""></script><script src="/component---src-pages-timeline-js-dcc741760f588e8f0200.js" async=""></script><script src="/f0d9b79f24a0985e3f3911e871f5374fd763bbdf-efc7f1ba0523bbcd09ad.js" async=""></script><script src="/0b10d7199d01ceeef934b1f7be1d4d848090e8b1-ce937dcc159ebdddc4a8.js" async=""></script><script src="/commons-47785431c9696da9c5f8.js" async=""></script><script src="/0f1ac474-cb5adb2ed9d11fe078e1.js" async=""></script><script src="/styles-407fe62976dc5310c43e.js" async=""></script><script src="/app-1ff00ff6183944037c30.js" async=""></script><script src="/framework-acd7498685eeb36e39da.js" async=""></script><script src="/webpack-runtime-f41cac52ff50071fe1f4.js" async=""></script></body></html>